src/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/Inlined/NullableDecimalAverageAggregationOperator.cs

```diff
@@ -118,8 +118,11 @@ protected override bool MoveNextCore(ref Pair<decimal, long> currentElement)
 
                     if (current.HasValue)
                     {
-                        sum += current.GetValueOrDefault();
-                        count++;
+                        checked
+                        {
+                            sum += current.GetValueOrDefault();
+                            count++;
+                        }
```

Comment:
Do you mean only checking the count for float/double/decimal?  I didn't do it that way initially because then it wouldn't match the non-nullable versions, so we'd probably want to change those in that case.
(I'm assuming you don't mean only checking the addition for long, which is the only thing that can overflow at the moment?  I debated that too, but it made me a little nervous, relying on the inner enumeration limit....)